encoding/thrift/procedure.go

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) 2016 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package thrift
+
+import (
+	"fmt"
+	"strings"
+)
+
+// procedureName gets the procedure name we should use for a Thrift method with
+// the given service and name.
+func procedureName(service, method string) string {
+	return fmt.Sprintf("%s::%s", service, method)
+}
+
+// splitProcedure splits the given procedure name into the Thrift service name
+// and method.
+func splitProcedure(proc string) (service, method string) {
+	parts := strings.SplitN(proc, "::", 2)
```

Comment:
rather than panic, might be better to handle the case where the split doesn't give you an array with 2 elements
```go
  switch {
  case len(parts) == 0:
    return "", ""
  case len(parts) == 1:
    return parts[0], ""
  default:
    return parts[0], parts[1]
  }
```